### ISSUE TITLE
Feature Request: Add geo location info to weather api

### ISSUE THREAD

**Issue by csm10495** [open-meteo/open-meteo#965]
Hey folks,

It would be helpful if there was a way to get a canonical location name for a given latitude and longitude.

For example, I use the API to fetch the local weather using lat/long though I don't have a good name for where the weather forecast is for.

I'd like to use this info to be able to link to a weather website to get more info on the forecast. Websites seem to generally be based on canonical locations instead of lat and long.

So I guess the request is for either a canonical location name or some other way of linking to a website to visually see the forecast the API gave me.

Thanks so much for your consideration.

**Comment by guidocioni** [comment]
> Hey folks,
> 
> It would be helpful if there was a way to get a canonical location name for a given latitude and longitude.
> 
> For example, I use the API to fetch the local weather using lat/long though I don't have a good name for where the weather forecast is for.
> 
> I'd like to use this info to be able to link to a weather website to get more info on the forecast. Websites seem to generally be based on canonical locations instead of lat and long.
> 
> So I guess the request is for either a canonical location name or some other way of linking to a website to visually see the forecast the API gave me.
> 
> Thanks so much for your consideration.

Getting a location from a pair of lat/lon coordinates (reverse geocoding) is actually not a trivial task because one needs to decide how to weight all the different geofeatures that are close to a point (city, town, address, location...).

There are many services that let you do that. I've been using the mapbox API (https://docs.mapbox.com/api/search/geocoding-v5/#reverse-geocoding) for many years without issues. There is a generous free tier that lets you try that out.

I'm pretty sure reverse geocoding is out of scope for openmeteo given the complexity. It is better to rely on an external service :)

**Comment by csm10495** [comment]
Yeah that makes complete sense. I guess an alternative would be to instead have a link to a corresponding webpage with the forecast. Then I could just link to that instead of trying to reverse it out.

If that is still outside of the scope: I understand and you can freely close this out. Thanks!

**Comment by terraputix** [comment]
You can configure a link with selected variables and coordinates (e.g. by searching a location in the location search) on the open-meteo.com/en/docs page, e.g. for Zurich https://open-meteo.com/en/docs/dwd-api?latitude=47.3667&longitude=8.55&hourly=temperature_2m,relative_humidity_2m
